public/components/trace_analytics/components/traces/span_detail_panel.tsx

```diff
@@ -55,6 +60,50 @@ export function SpanDetailPanel(props: {
   } else {
     [data, setData] = [localData, localSetData];
   }
+  const fullRange = [0, data.ganttMaxX * 1.1];
+  const [selectedRange, setSelectedRange] = useState(fullRange);
+  const isLocked = useObservable(chrome!.getIsNavDrawerLocked$() ?? false);
+  const [isFullScreen, setIsFullScreen] = useState(false);
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const [availableWidth, setAvailableWidth] = useState<number>(window.innerWidth);
+
+  useEffect(() => {
+    const updateAvailableWidth = () => {
+      if (containerRef.current) {
+        setAvailableWidth(containerRef.current.getBoundingClientRect().width);
+      } else {
+        setAvailableWidth(window.innerWidth);
+      }
+    };
+
+    const handleFullScreenChange = () => {
+      const isFullscreenActive = !!document.fullscreenElement;
+      setIsFullScreen(isFullscreenActive);
+      updateAvailableWidth();
+    };
+
+    // Add event listeners for window resize and full-screen toggling
+    window.addEventListener('resize', updateAvailableWidth);
+    document.addEventListener('fullscreenchange', handleFullScreenChange);
+
+    // Initial update
+    updateAvailableWidth();
+
+    return () => {
+      // Clean up event listeners
+      window.removeEventListener('resize', updateAvailableWidth);
+      document.removeEventListener('fullscreenchange', handleFullScreenChange);
+    };
+  }, []);
+
+  const dynamicLayoutAdjustment = useMemo(() => {
+    return isLocked ? availableWidth - 350 : availableWidth - 150;
```

Comment:
Should this change when new home is enabled/disabled? Is the pixel width of the sidebar constant with and without the new home?

Reply:
I don't think it is necessary as both are filling the window properly. Can if needed but I think it would just be a roughly 5-10 pixel change that wouldn't be noticeable.
Old navigation:

https://github.com/user-attachments/assets/3506615f-ae4f-4b81-9dde-d91982ac996a

New navigation:

https://github.com/user-attachments/assets/9d1a7aa5-0239-45e9-8e42-5fed2c38a175